packages/itwin/grouping-mapping-widget/src/widget/components/GroupPropertyAction.tsx

```diff
@@ -264,90 +97,80 @@ const GroupPropertyAction = ({
   keySet,
   returnFn,
 }: GroupPropertyActionProps) => {
-  const iModelConnection = useActiveIModelConnection() as IModelConnection;
+  const iModelConnection = useActiveIModelConnection();
   const { getAccessToken } = useGroupingMappingApiConfig();
   const mappingClient = useMappingClient();
   const [propertyName, setPropertyName] = useState<string>("");
   const [dataType, setDataType] = useState<DataType>(DataType.Undefined);
   const [quantityType, setQuantityType] = useState<QuantityType>(QuantityType.Undefined);
-  const [classToPropertiesMapping, setClassToPropertiesMapping] =
-    useState<Map<string, Property[]>>();
-  const [ecProperties, setEcProperties] = useState<ECProperty[]>(
-    []
-  );
+  const [selectedProperties, setSelectedProperties] = useState<PropertyMetaData[]>([]);
+  const [propertiesMetaData, setPropertiesMetaData] = useState<PropertyMetaData[]>([]);
+  const [propertiesNotFoundAlert, setPropertiesNotFoundAlert] = useState<boolean>(false);
   const [validator, showValidationMessage] = useValidator();
-  const [propertyAlert, setPropertyAlert] = useState<boolean>(false);
   const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [searchInput, setSearchInput] = useState<string>("");
+  const [activeSearchInput, setActiveSearchInput] = useState<string>("");
+  const [searched, setSearched] = useState<boolean>(false);
+  const [activeDragProperty, setActiveDragProperty] = useState<PropertyMetaData>();
+  const sensors = useSensors(
+    useSensor(PointerSensor),
+    useSensor(KeyboardSensor, {
+      coordinateGetter: sortableKeyboardCoordinates,
+    })
+  );
+  const [showModal, setShowModal] = useState<boolean>(false);
 
-  useEffect(() => {
-    const getContent = async () => {
-      setIsLoading(true);
-      const ruleSet: Ruleset = {
-        id: "element-properties",
-        rules: [
-          {
-            ruleType: RuleTypes.Content,
-            specifications: [
-              {
-                specType: ContentSpecificationTypes.SelectedNodeInstances,
-              },
-            ],
-          }],
-      };
-      const requestOptions: ContentDescriptorRequestOptions<IModelConnection, KeySet, RulesetVariable> = {
-        imodel: iModelConnection,
-        keys: keySet,
-        rulesetOrId: ruleSet,
-        displayType: DefaultContentDisplayTypes.PropertyPane,
-      };
-      const content = await Presentation.presentation.getContentDescriptor(
-        requestOptions
-      );
+  const handleDragStart = (event: DragStartEvent) => {
```

Comment:
This might need to be a callback, because it depends on `selectedProperties` value.

Reply:
Both needed callbacks actually, the `onHandle` props most likely have these functions as dependencies as well.